frontend/javascripts/admin/api_flow_types.js

```diff
@@ -378,22 +378,37 @@ export type DatasetConfig = {
   +zipFile: File,
 };
 
-type WkConnectLayer = {
+type NeuroglancerLayer = {
   // This is the source URL of the layer, should start with gs://, http:// or https://
   source: string,
   type: "image" | "segmentation",
 };
 
-export type WkConnectDatasetConfig = {
-  neuroglancer: {
-    [organizationName: string]: {
-      [datasetName: string]: {
-        layers: { [layerName: string]: WkConnectLayer },
-      },
+type NeuroglancerDatasetConfig = {
+  [organizationName: string]: {
+    [datasetName: string]: {
+      layers: { [layerName: string]: NeuroglancerLayer },
+    },
+  },
+};
+
+type BossDatasetConfig = {
+  [organizationName: string]: {
+    [datasetName: string]: {
+      domain: string,
+      collection: string,
+      experiment: string,
+      username: string,
+      password: string,
     },
   },
 };
 
+export type WkConnectDatasetConfig = {
+  neuroglancer?: NeuroglancerDatasetConfig,
+  boss?: BossDatasetConfig,
```

Comment:
The type suggests that neuroglancer and boss could be defined at the same time. Can this really happen? Otherwise, maybe we can use `NeuroglancerDatasetConfig | BossDatasetConfig`?

Reply:
Ah, it's probably due to backwards compatibility, right? The current way is also fine with me!

Reply:
The reason is actually that wkConnect expects exactly this kind of object as a dataset configuration, usually with either a neuroglancer or a boss key.